pkg/bloomcompactor/metrics.go

```diff
@@ -0,0 +1,92 @@
+package bloomcompactor
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+const (
+	metricsNamespace = "loki_bloomcompactor"
+)
+
+type metrics struct {
+	compactionRunsStarted          prometheus.Counter
+	compactionRunsCompleted        prometheus.Counter
+	compactionRunsErred            prometheus.Counter
+	compactionRunDiscoveredTenants prometheus.Gauge
+	compactionRunSkippedTenants    prometheus.Gauge
+	compactionRunSucceededTenants  prometheus.Gauge
+	compactionRunFailedTenants     prometheus.Gauge
+	compactionRunSkippedJobs       prometheus.Gauge
+	compactionRunSucceededJobs     prometheus.Gauge
+	compactionRunFailedJobs        prometheus.Gauge
```

Comment:
all these metrics should be the counters.
we can use `rate` , `increase` . otherwise, we might miss relevant data if the value changes up and down between the scraping.

Reply:
Done, thanks for the tip!

Reply:
I'd use counters for definitives ones like err, success, skipped etc once the job is completed
but I'd argue gauge is appropriate only to record ongoing jobs (CompactionRunning) 
I can see how many compactions are running at the same time, but the gauge will drop once it's completed and record a success, fail etc counter.